app/src/main/java/com/capstone/weatherapp/WeatherRepository.kt

```diff
@@ -26,23 +26,40 @@ class WeatherRepository(private val db: WeatherDatabase) {
 
     suspend fun refreshCityList() {
         withContext(Dispatchers.IO) {
+            _cityListData.postValue(State.Loading)
```

Comment:
you won't need to use postValue here once you update _cityListData to be a var instead of a val. but good call to set it to loading 👍